beetsplug/mosaic.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2015-2016, Ohm Patel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+"""Allows beets to create a mosaic from covers."""
+from __future__ import division, absolute_import, print_function
+
+import os.path
+
+from beets.plugins import BeetsPlugin
+from beets import ui
+from beets.ui import decargs
+from beets.util import syspath, normpath, displayable_path, bytestring_path
+from beets.util.artresizer import ArtResizer
+from beets import config
+from beets import art
+from PIL import Image
+
+import math
+
+
+class MosaicCoverArtPlugin(BeetsPlugin):
+	col_size = 4
+	margin = 3
+
+	def __init__(self):
+		super(MosaicCoverArtPlugin, self).__init__()
+		self.config.add({'maxwidth': 300})
+		self.maxwidth = self.config['maxwidth'].get(int)
+
+ 	def commands(self):
+		cmd = ui.Subcommand('mosaic', help=u"create mosaic from coverart")
+
+		def func(lib, opts, args):
+			self._generate_montage(lib, lib.albums(ui.decargs(args)), u'mos.png')

```

Comment:
Should the mosaic filename be chosen by the user, at least optionally? It makes sense from a ux point of view.
Also, this would make the code more sensible, as for now `_generate_montage` has the `output_fn` parameter which is always hardcoded to `u'mos.png'`. That made me scratch my head a little bit.

`embedart` for example could be an inspiration for this. Moreover, it  also does not let the user choose the extension (here I suppose you only generate `png` for convenience reasons), so you could use the exact same convention and ux.